enumerables.rb

```diff
@@ -0,0 +1,153 @@
+module Enumerable
+  def my_each
+    return to_enum :name unless block_given?
+
+    for element in self
+      yield element
+    end
+  end
+
+  def my_each_with_index
+    return to_enum :name unless block_given?
+
+    c = 0
+    for element in self
+      yield element, c
+      c += 1
+    end
+  end
+
+  def my_any?(args = nil)
+    if args.nil?
+      if block_given?
+        my_each { |x| return true if yield(x) }
+      else
+        my_each { |x| return true if x }
+      end
+    elsif args.is_a?(Regexp)
+      my_each { |x| return true if x.match(args) }
+    elsif args.is_a?(Module)
+      my_each { |x| return true if x.is_a?(args) }
+    else
+      my_each { |x| return true if x == args }
+    end
+    false
+  end
+
+  def my_none?(args = nil)
+    if args.nil?
+      if block_given?
```

Comment:
At the my_none?, we got these errors:

1. rspec ./spec/enumerables_spec.rb:330 # enumerables #my_none? when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
2. rspec ./spec/enumerables_spec.rb:333 # enumerables #my_none? when a pattern other than Regex or a Class is given returns false only if one of the collection matches the pattern